api/client/client.go

```diff
@@ -2265,6 +2265,27 @@ func (c *Client) GetInstallers(ctx context.Context) ([]types.Installer, error) {
 	return installers, nil
 }
 
+// GetUIConfig gets the configuration for the UI served by the proxy service
+func (c *Client) GetUIConfig(ctx context.Context) (types.UIConfig, error) {
+	resp, err := c.grpc.GetUIConfig(ctx, &emptypb.Empty{}, c.callOpts...)
+	return resp, trail.FromGRPC(err)
+}
+
+// SetUIConfig sets the configuration for the UI served by the proxy service
+func (c *Client) SetUIConfig(ctx context.Context, uic types.UIConfig) error {
+	uicV1, ok := uic.(*types.UIConfigV1)
```

Comment:
AIUI casting an interface like this is a sign something is amiss.... as this is the first time this is being created shouldn't this just be a V1 type?

Reply:
This is common with how we handle Teleport resources - the type from the `types` package is an interface, and we convert to the underlying implementation before invoking the GRPC method. A grep for `.(*` in this same file will uncover plenty of examples.